src/lib.rs

```diff
@@ -285,4 +378,24 @@ mod tests {
 
         Ok(())
     }
+
+    fn number_digit_addition_prop(a: &[u32], b: impl Into<Digit>) -> Result<(), TestCaseError> {
+        let b: Digit = b.into();
+        // We assume that `a` is non-negative
+        prop_assume!(a.is_empty() || (a.last().unwrap() >> 31) == 0);
+
+        let a_gmp = BigInt::from_digits_u32(a);
+        let result_gmp = &a_gmp + b;
+        let limbs_gmp = result_gmp.to_digits_u32();
+
+        let a_num = num_bigint::BigUint::new(a.to_vec());
+        let b_u32_vec = digits_to_limbs_u32(&[b]).to_vec();
```

Comment:
Hmm. I think we can remove this conversion:

```rust
fn number_digit_addition_prop<D>(a: &[u32], b: D) -> Result<(), TestCaseError> 
where 
    BigInt: std::ops::Add<D>,
    num_bigint::BigUint: std::ops::Add<D>
{
    // We assume that `a` is non-negative
    prop_assume!(a.is_empty() || (a.last().unwrap() >> 31) == 0);

    let a_gmp = BigInt::from_digits_u32(a);
    let result_gmp = &a_gmp + b;
    let limbs_gmp = result_gmp.to_digits_u32();

    let a_num = num_bigint::BigUint::new(a.to_vec());
    let result_num = a_num + b;

    // ...
}
```

Reply:
I just think that we should not get `digits_to_limbs_u32` involved to this particular test

Reply:
I tried to implement `number_digit_addition_prop` without any casting from `D` to anything else, and couldn't find a normal way to do it. The problem is, `num_bigint` knows nothing about `Digit`s and so we cannot really demand any interaction between them. I opted out for a pretty ugly solution of supplying an extra value of `b` as a `u32` to be able to add it to a `BigUInt` value. Please let me know if you think of anything better.